pkg/build/generator/generator_test.go

```diff
@@ -2203,8 +2152,8 @@ func TestOverrideSourceStrategyIncrementalOption(t *testing.T) {
 			},
 		}, nil
 	}
+	g.Client = client
```

Comment:
The fact that people screwed up (missed) this reassignment was the motivation to move to a pattern that was less error prone.

I'd be happy with introduce a mockBuildGeneratorDefault() function and a mockBuildGeneratorCustom(func1,func2,func3,func4,etc) signature, but I do prefer the model where you create the generator with your customized functions over creating a generator and then doing the right dance to replace the functions.

Reply:
I'd also be amenable to changing the way the "defaulting" of the functions is handled such that it's all within the mockBuildGeneratorCustom() function, rather than the separate helper functions that were introduced for defaulting each function, but that's really neither here nor there to me.